examples/nlp/language_modeling/megatron_gpt_eval.py

```diff
@@ -12,166 +12,62 @@
+from nemo.collections.nlp.modules.common.text_generation_utils import generate
+from nemo.collections.nlp.modules.common.transformer.text_generation import LengthParam, SamplingParam
 from nemo.collections.nlp.parts.nlp_overrides import NLPDDPPlugin
-from nemo.utils import logging
+from nemo.core.config import hydra_runner
 from nemo.utils.app_state import AppState
 from nemo.utils.model_utils import inject_model_parallel_rank
 
-"""
-Usage:
-    a. If you need to run model on a few prompts from the file:
-        python megatron_gpt_eval.py \
-            --model_file=PATH_TO_MODEL \
-            --path_to_file=PATH_TO_FILE \
-            --tokens_to_generate=32 \
-            --batch_size=16 \
-
-    b. If you need to run model on a prompt from the CLI:
-        python megatron_gpt_eval.py \
-            --model_file=PATH_TO_MODEL \
-            --tokens_to_generate=32 \
-            --prompt=YOUR_PROMPT
-
-    c. If you don't need to generate tokens and need model to compute logprobs:
-         python megatron_gpt_eval.py \
-            --model_file=PATH_TO_MODEL \
-            --path_to_file=PATH_TO_FILE \
-            --batch_size=16 \
-            --compute_logprobs=True \
-            --prompt .
-
-    d. If you need to run a prompt-tuned model on a few prompts from a file:
-        python megatron_gpt_eval.py \
-            --use_soft_prompts \
-            --model_file=PATH_TO_MODEL \
-            --path_to_file=PATH_TO_FILE \
-            --tokens_to_generate=32 \
-            --batch_size=16 \
-
-        The path_to_file containing the model prompts should be a json with prompts in the format:
-            {'prompt_tag': tag1, 'text': prompt1}
-            {'prompt_tag': tag1, 'text': prompt2}
-            {'prompt_tag': tag3, 'text': prompt3}
-
-    e. If you need to run a prompt-tuned model on a prompt from the CLI:
-        python megatron_gpt_eval.py \
-            --use_soft_prompts \
-            --model_file=PATH_TO_MODEL \
-            --tokens_to_generate=32 \
-            --prompt_tag=PROMPT_TAG_STRING \
-            --prompt=YOUR_PROMPT
-"""
-
-assert torch.cuda.is_available()
-
-
-def main():
-    parser = ArgumentParser()
-
-    # args for loading the model, either from .nemo file or from PTL checkpoint
-    parser.add_argument("--model_file", type=str, default="", required=False, help="Pass path to model's .nemo file")
-    parser.add_argument(
-        "--checkpoint_dir",
-        type=str,
-        default=None,
-        required=False,
-        help="If not using a .nemo file. Path to PTL checkpoints saved during training. Ex: /raid/nemo_experiments/megatron_gpt/checkpoints",
-    )
-    parser.add_argument(
-        "--checkpoint_name",
-        type=str,
-        default=None,
-        required=False,
-        help="If not using a .nemo file. Name of checkpoint to be used. Ex: megatron_gpt--val_loss=6.34-step=649-last.ckpt",
-    )
+try:
+    from apex.transformer import parallel_state
```

Comment:
Can we add back a docstring with usage examples?

Reply:
Good point. will add a docstring.